src/input/mpegts/iptv/iptv_http.c

```diff
@@ -347,6 +347,69 @@ iptv_http_reconnect ( http_client_t *hc, const char *url )
   urlreset(&u);
 }
 
+/*
+ * Return the index of the last occurrence of character `x` in string `s`
+ */
+static
+int last_index_of(const char *s, char x) {
+  int i;
+  int len = strlen(s);
+  assert(s != NULL);
+  for (i = len - 1; i >= 0; i--) {
+    if (s[i] == x) {
+      return i;
+    }
+  }
+  return -1; // x not found
+}
+
+/*
+ * Allocates a string with the absolute URL obtained by merging the absolute base and the relative URLs
+ */
+static char * merge_absolute_and_relative_urls(const char * absolute_url, const char * relative_url) {
+  size_t absoute_key_url_len, last_slash_pos;
+  char * absolute_key_url;
+  const unsigned int max_levels = 100;
+  int i;
+
+  // allocate enough memory to contain `s` + `url` + \0
+  absoute_key_url_len = strlen(absolute_url) + strlen(relative_url) + 1;
+  absolute_key_url = (char*)malloc(absoute_key_url_len);
+  memset(absolute_key_url, 0, absoute_key_url_len);
+
+  // copy the url up to the last /
+  // http://test/ab/cd becomes http://test/ab/
+  last_slash_pos = last_index_of(absolute_url, '/');
+  assert(last_slash_pos < absoute_key_url_len);
+  assert(last_slash_pos > 0);
+  memcpy(absolute_key_url, absolute_url, last_slash_pos+1);
+
+  for (i = 0; i < max_levels && (memcmp(relative_url, "../", strlen("../")) == 0); i++) {
```

Comment:
Is it possible, that `memcmp` reads beyond the end of the `relative_url` string here? Shouldn't strncmp be used here?